open_fdd/quality.py

```
@@ -0,0 +1,330 @@
+    valid = valid & ~non_num
+
+    sent = _is_sentinel(num, sentinels)
+    reason = reason.mask(sent, REASON_SENTINEL)
+    valid = valid & ~sent
+
+    lo_hi = bounds if bounds is not None else ROLE_BOUNDS.get(family)
+    if lo_hi is not None:
+        lo, hi = lo_hi
+        scaled = _scale_position(num)
+        if family in POSITION_ROLES or any(k in family for k in ("valve", "damper", "speed")):
+            # Accept 0–1 or 0–100
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)